File: output/剑指Offer32-I.从上到下打印二叉树/剑指Offer32-I-从上到下打印二叉树.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[int]:
        if not root:
            return []
        res = []
        q = deque([root])
        while q:
            now = q.popleft()
            res.append(now.val)
            if now.left:
                q.append(now.left)
            if now.right:
                q.append(now.right)
        return res

File: 答题/标签/广度优先搜索/剑指 Offer 32 - I.从上到下打印二叉树/剑指 Offer 32 - I-从上到下打印二叉树.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) []int {
    if root == nil{
        return []int{}
    }
    res := []int{}
    l := []*TreeNode{root}
    for len(l) != 0{
        newL := []*TreeNode{}
        for i := 0; i < len(l); i++{
            res = append(res, l[i].Val)
            if l[i].Left != nil{
                newL = append(newL, l[i].Left)
            }
            if l[i].Right != nil{
                newL = append(newL, l[i].Right)
            }
        }
        l = newL
    }
    return res

}